test/common.js

```diff
@@ -25,20 +27,60 @@ exports.isWindows = () => {
   return process.platform === 'win32';
 };
 
-exports.validate = (t, report, pid) => {
+exports.validate = (t, report, options) => {
   t.test('Validating ' + report, (t) => {
     fs.readFile(report, (err, data) => {
+      const pid = options ? options.pid : process.pid;
       const reportContents = data.toString();
-      const plan = REPORT_SECTIONS.length + (pid ? 1 : 0);
+      const nodeComponents = Object.keys(process.versions);
+      const expectedVersions = options ?
+                               options.expectedVersions || nodeComponents :
+                               nodeComponents;
+      const plan = REPORT_SECTIONS.length + nodeComponents.length + 2;
       t.plan(plan);
-      if (pid) {
-        t.match(reportContents, new RegExp('Process ID: ' + pid),
-                'Checking report contains expected process ID ' + pid);
-      }
+
+      // Check all sections are present
       REPORT_SECTIONS.forEach((section) => {
         t.match(reportContents, new RegExp('==== ' + section),
                 'Checking report contains ' + section + ' section');
       });
+
+      // Check NodeReport section
+      const nodeReportSection = getSection(reportContents, 'NodeReport');
+      t.match(nodeReportSection, new RegExp('Process ID: ' + pid),
+              'NodeReport section contains expected process ID');
+      if (options && options.expectNodeVersion === false) {
+        t.match(nodeReportSection, /Unable to determine Node.js version/,
+                'NodeReport section contains expected Node.js version');
+      } else {
+        t.match(nodeReportSection,
+                new RegExp('Node.js version: ' + process.version),
+                'NodeReport section contains expected Node.js version');
+      }
+      nodeComponents.forEach((c) => {
+        if (c !== 'node') {
+          if (expectedVersions.indexOf(c) === -1) {
```

Comment:
NodeReport claims to be Node.js v4.x compatible (`"node": ">=4.0.0"`) and the version of V8 in v4.x doesn't have `Array.prototype.includes()`.